include/onnxruntime/core/framework/tensor.h

```diff
@@ -164,8 +164,15 @@ class Tensor final {
     shape_ = new_shape;
   }
 
+  //return zero if integer overflow
   size_t Size() const noexcept {
```

Comment:
Would it be better to throw an exception if there's an error calculating the size? I assume you don't expect that to happen, and returning 0 could accidentally be interpreted as an empty tensor, delaying when the error due to this breaks something (making it much harder to debug).

Reply:
The function is marked as 'noexcept'. If I change it to throw exception, I need to change a lot different places.

Reply:
Why? I thought noexcept was mainly an optimization opportunity and didn't require changes at the calling site. Suddenly returning zero seems like it would actually need all the calling sites to be aware of that possibility and handle it, so I would have thought throwing would have less impact on other code. 

---
In reply to: [255722659](https://github.com/Microsoft/onnxruntime/pull/465#discussion_r255722659) [](ancestors = 255722659)